### PR TITLE
[LLVMGPU] Add basic lowering pipeline without tiling and distribution

### PR DESCRIPTION
**Pull request by jinchen62**
Refer to #14377

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]: leftover debug statement

File: compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.td
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.td:32]: Add this as a new value at the bottom (110) rather than incrementing all of the other pipeline enums.

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:1532]: Can you add a test to `LLVMGPU/test/llvmgpu_pipeline_test.mlir` (or some place like that) showing what dispatch this is solving? Is this just for scalar dispatches?

**Review by qedawkins — CHANGES_REQUESTED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp
Comment (jinchen62) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:1532]: I was thinking the test would be basically the same as https://github.com/openxla/iree/pull/14377/files#diff-24b6180b258e5500ab5f5385e05b06095ac5fb6c28bf469369ec38d5b519c145, so was not sure if I should add it.

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp
Comment (antiagainst) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:1]: Could you also do the same in `ROCDLKernelConfig.cpp` too? That's the new pass we are developing for ROCDL.

File: compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPULowerExecutableTarget.cpp
Comment (antiagainst) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPULowerExecutableTarget.cpp:1]: Similarly here could you also add it to `ROCDLLowerExecutableTarget.cpp`?

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp
Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:1532]: Yes let's add a test like the above into `LLVMGPU/test/ROCDL` and use `iree-rocdl-lower-executable-target-pass`.

File: compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.td
Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.td:32]: This is fine. We don't need to preserve the numeric value of the passes anyway. This actually gives us a more logical order.

**Review by antiagainst — CHANGES_REQUESTED** [review]
Nice, thanks for fixing this!

File: compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLLowerExecutableTarget.cpp
Comment (antiagainst) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLLowerExecutableTarget.cpp]: Use `CodeGenPipeline` here like at L54 to be shorter.

File: compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLSelectLoweringStrategy.cpp
Comment (antiagainst) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLSelectLoweringStrategy.cpp:25]: Is this needed actually? Don't think we involve transform dialect stuff in this pull request?

**Review by antiagainst — CHANGES_REQUESTED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLSelectLoweringStrategy.cpp
Comment (jinchen62) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLSelectLoweringStrategy.cpp:25]: I got
> LLVM ERROR: Building op `bufferization.alloc_tensor` but it isn't known in this MLIRContext: the dialect may not be loaded or this operation hasn't been added by the dialect.

without adding it.

Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLSelectLoweringStrategy.cpp:25]: You need the bufferization dialect. Not the transform dialect.

Comment (jinchen62) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLSelectLoweringStrategy.cpp:25]: @MaheshRavishankar Thanks!

**Review by qedawkins — APPROVED** [review]
My comments were addressed so don't want to block. Wait for approval from Lei first though.

**Review by antiagainst — APPROVED** [review]

**Comment by antiagainst** [comment]
Ah, looks this breaks CUDA tests.. https://github.com/openxla/iree/actions/runs/8011433067/job/21885181762 @jinchen62 could you revert the commit for now? I can look into fix and reland later.

**Pull request merged**
